core/creators/themedui.js

```diff
@@ -384,6 +389,61 @@ CKEDITOR.replaceClass = 'ckeditor';
 		return editor;
 	}
 
+	// Delay editor creation if given element is detached from DOM.
+	//
+	// Require `config.delayIfDetached` to be `true`.
+	//
+	// If `config.delayIfDetached_callback` is declared as function, it will be invoked with single argument:
+	// function, that should be called to create editor.
+	//
+	// Otherwise: periodically (with `setInterval`) run check if element is attached to DOM and create editor automatically.
+	//
+	// ```js
+	//	CKEDITOR.replace( detachedEditorElement, {
+	//		delayIfDetached: true,
+	//		delayIfDetached_interval: 50,
+	//		delayIfDetached_callback: registerCallback
+	//		}
+	//	} );
+	// ```
+	//
+	// @param {CKEDITOR.element} element The DOM element.
+	// @param {String} mode CKEditor mode (source/wysiwyg).
+	// @param {Object} [config] The specific configuration to apply to this
+	// editor instance. Configuration set here will override the global CKEditor settings.
+	// @param {String} [data] Initial value for the instance.
+	function delayCreationOnDetachedElement( element, mode, config, data ) {
+		if ( !config || !config.delayIfDetached || !element.isDetached() ) {
+			return false;
+		}
+
+		if ( config.delayIfDetached_callback ) {
+			config.delayIfDetached_callback( function() {
+				createInstance( element, config, data, mode );
+			} );
+
+			return true;
+		}
```

Comment:
In this scenario we don't inform the user that the editor creation was suspended 🤔  Let's add a `CKEDITOR.warn()` call here too. I was wondering about the consistency in both cases (callback vs interval) and I found out the best solution is adding a `mode` parameter to the  'editor-delayed-creation' error (we already have it), then in callback scenario have `mode: 'callback'` and in interval one `mode: 'interval - ' + interval + ' ms'`. The code error info is rather hard to use programatically so I don't think it's super useful and important to keep it a numeric value instead of such a string. WDYT?

Reply:
> In this scenario we don't inform the user that the editor creation was suspended

I assume, that we don't have to inform the user if a callback was provided intentionally, but a simple warn should be helpful in debugging. I'll add another message :) I think such distinction also would be useful. The `mode` option sounds reasonable. TBH I would separate mode and time value as another property, but as you said, it should be used only for information purposes - so I leave it as once :)

Reply:
Actually, I added one more warn ;p To preserve consistent behavior in `interval` and `callback` mode, I added `editor-delayed-creation-sucess` to function which is passed as an argument to the user callback. So now, we have a warning if creation is delayed (both cases) and a warning for successful creation (both cases).